planner/core/logical_plan_builder.go

```diff
@@ -1994,6 +1998,46 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 	return result, nil
 }
 
+func (b *PlanBuilder) buildDataSourceFromView(dbName model.CIStr, tableInfo *model.TableInfo) (LogicalPlan, error) {
+	var (
+		selectNode        ast.StmtNode
+		selectLogicalPlan Plan
+		err               error
+	)
+	charset, collation := b.ctx.GetSessionVars().GetCharsetInfo()
+	selectNode, err = parser.New().ParseOneStmt(tableInfo.View.SelectStmt, charset, collation)
+	if err != nil {
+		return nil, err
+	}
+	selectLogicalPlan, err = b.Build(selectNode)
+	if err != nil {
+		return nil, err
+	}
+	projSchema := expression.NewSchema(make([]*expression.Column, 0, len(tableInfo.View.Cols))...)
+	projUponCols := make([]*expression.Column, 0, len(tableInfo.View.Cols))
```

Comment:
1. s/ projUponCols/ projExprs
2. can we define projExprs as []*expression.Expression?

Reply:
fine,done